cf/src/main/scala/com/timeout/scalacloudformation/Encoding.scala

```diff
@@ -43,30 +55,110 @@ object Encoding {
+        case String => "String"
+        case Number => "Number"
+        case `List<Number>` => "List<Number>"
+        case CommaDelimitedList => "CommaDelimitedList"
+        case t: AwsListType => s"List<${awsEnum.encode(t.tpe)}>"
+        case t: AwsType => awsEnum.encode(t.tpe)
+      }
+      Json.fromString(s)
+    }
   }
 
-  implicit def encodeResource[T <: Resource, LabelledGeneric, R <: HList](
-    implicit
-    gen: LabelledGeneric.Aux[T, R],
-    encode: ObjectEncoder[R]
-  ): Encoder[T] = Encoder.instance[T] { r =>
-    Json.obj(
-      r.logicalId -> Json.obj(
-        "Type" -> Json.fromString(r.fqn),
-        "Properties" -> encode(gen.to(r)).asJson
+  implicit val autoscalingPolicy: Encoder[AutoscalingCreationPolicy] =
+    deriveEncoder[AutoscalingCreationPolicy]
+
+  implicit val resourceSignal: Encoder[ResourceSignal] =
+    deriveEncoder[ResourceSignal]
+
+  implicit val creationPolicy: Encoder[CreationPolicy] =
+    deriveEncoder[CreationPolicy]
+
+  implicit val autoScalingReplacingUpdate: Encoder[AutoScalingReplacingUpdate] =
+    deriveEncoder[AutoScalingReplacingUpdate]
+
+  implicit val autoScalingRollingUpdate: Encoder[AutoScalingRollingUpdate] =
+    deriveEncoder[AutoScalingRollingUpdate]
+
+  implicit val autoScalingScheduledAction: Encoder[AutoScalingScheduledAction] =
+    deriveEncoder[AutoScalingScheduledAction]
+
+  implicit val updatePolicy: Encoder[UpdatePolicy] =
+    deriveEncoder[UpdatePolicy]
+
+  implicit val encodeParam: Encoder[Parameter] =
+    Encoder.instance[Parameter] { p =>
+      val common = Json.obj(
+        "Type" -> p.Type.asJson,
+        "Description" -> p.Description.asJson,
+        "NoEcho" -> p.NoEcho.asJson
       )
+
+      val specific = p match {
+        case sp: Parameter.Str =>
+          Json.obj(
+            "MaxLength" -> sp.MaxLength.asJson,
+            "MinLength" -> sp.MinLength.asJson,
+            "AllowedPattern" -> sp.AllowedPattern.asJson,
+            "AllowedValues" -> sp.AllowedValues.asJson,
+            "Default" -> sp.Default.asJson
+          )
+        case np: Parameter.Number =>
+          Json.obj(
+            "MaxValue" -> np.MaxValue.asJson,
+            "MinValue" -> np.MinValue.asJson,
+            "Default" -> np.Default.asJson
+          )
+        case p: CommaDelimited =>
+          Json.obj(
+            "AllowedValues" -> p.AllowedValues.map(_.mkString(",")).asJson,
+            "Default" -> p.Default.map(_.mkString(",")).asJson
+          )
+        case _ => Json.obj()
+      }
+
+      Json.obj(p.logicalId -> common.deepMerge(specific))
+    }
+
+  implicit val condition: Encoder[Condition] = deriveEncoder[Condition]
+
+  implicit val output: Encoder[Output] = Encoder.instance { o =>
+    Json.obj(o.logicalId -> Json.obj(
+      "Description" -> o.Description.asJson,
+      "Value" -> o.Value.asJson,
+      "Condition" -> o.Condition.map(_.logicalId).asJson,
+      "Export" -> o.Export.asJson //TODO: support cross-stack output
```

Comment:
It looks like the name is always wrapped into a `Name` key:

```javascript
"Outputs" : {
  "Logical ID" : {
    "Description" : "Information about the value",
    "Value" : "Value to return",
    "Export" : {
      "Name" : "Value to export"
    }
  }
}
```

http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html